cuda_phase-retrieval_class/test/cupy_caster.hpp

```diff
@@ -44,6 +44,15 @@ namespace pybind11 { namespace detail {
 					return false;
             }
 
+            if(src.attr("shape").cast<vector<size_t>>().size() != TDim)
```

Comment:
Add a test to check, if `TDim` is `0`. If `TDim is `0`, skip the shape test, because `0` means the dimension can be flexible